llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -399,39 +398,30 @@ TraverseCGToFindSPIRKernels(const Function *StartingFunction) {
     }
   }
 
-  return {false, std::move(KernelNames)};
+  return std::move(KernelNames);
 }
 
 std::vector<StringRef> getKernelNamesUsingAssert(const Module &M) {
-  Optional<const Function *> DevicelibAssertFailFunction;
-  std::vector<StringRef> SPIRKernelNames;
-  // This loop finds all SPIR kernel's names and __devicelib_assert_fail
-  // function if it is present.
-  for (const Function &F : M) {
-    if (F.getCallingConv() == CallingConv::SPIR_KERNEL)
-      SPIRKernelNames.push_back(F.getName());
-
-    if (F.getName().startswith("__devicelib_assert_fail")) {
-      assert(!DevicelibAssertFailFunction.hasValue());
-      DevicelibAssertFailFunction = &F;
-    }
-  }
-
+  auto DevicelibAssertFailFunction = M.getFunction("__devicelib_assert_fail");
   if (!DevicelibAssertFailFunction)
     return {};
 
   auto TraverseResult =
-      TraverseCGToFindSPIRKernels(*DevicelibAssertFailFunction);
-  std::vector<StringRef> Result;
-  if (TraverseResult.first) {
-    // If assert is met in some indirectly callable function than
-    // we return all kernels in Module due to the current assert's design.
-    Result = std::move(SPIRKernelNames);
-  } else {
-    Result = std::move(TraverseResult.second);
+      TraverseCGToFindSPIRKernels(DevicelibAssertFailFunction);
+
+  if (TraverseResult.hasValue()) {
+    return std::move(*TraverseResult);
+  }
```

Comment:
Please, keep code style consistent and aligned with LLVM coding style guide.
```suggestion
  if (TraverseResult.hasValue())
    return std::move(*TraverseResult);
```